esmvalcore/cmor/_fixes/native6/era5.py

```diff
@@ -433,3 +569,16 @@ def fix_metadata(self, cubes):
             fixed_cubes.append(cube)
 
         return fixed_cubes
+
+    def fix_data(self, cube):
+        """Fix data."""
+        regridding_enabled = self.extra_facets.get('regrid', True)
+        if regridding_enabled and has_unstructured_grid(cube):
+            logger.debug(
+                "Automatically regrid ERA5 data (variable %s) to %s° "
+                "grid using bilinear regridding",
+                self.vardef.short_name,
+                DEFAULT_ERA5_GRID,
+            )
+            cube = regrid(cube, DEFAULT_ERA5_GRID, 'linear')
```

Comment:
While I can see that the feature is useful, in particular to make seamlessly switching between NetCDF and GRIB ERA5 data possible, I'm not not sure if it should be enabled by default: I just checked a few recipes what the effect of this would be:
```
clouds/recipe_lauer22jclim_fig5_lifrac.yml
clouds/recipe_lauer22jclim_fig1_clim.yml
recipe_climwip_test_performance_sigma.yml
bock20jgr/recipe_bock20jgr_fig_1-4.yml
ipccwg1ar6ch3/recipe_ipccwg1ar6ch3_fig_3_19.yml
model_evaluation/recipe_model_evaluation_basics.yml
```
and in all cases it leads to double regridding. Therefore I think that disabled by default may be a better option.

Another concern is that a fix seems conceptually the wrong place to implement regridding, as this starts mixing preprocessing with data loading. But maybe there is no way around it.

Reply:
It would also be possible to automatically add a regrid preprocessor in `esmvalcore/_recipe/recipe.py` for ERA5 data, but I'm not sure if that would be any cleaner.

Reply:
There is the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) and to me at least, automatic regridding for one dataset and not for any other dataset is rather surprising, even if it may make sense in this case.

Reply:
One could also argue that the least surprise would be to get files on a regular grid (like the ones you get when you download the netCDF data).

How about checking the extension of the files in `esmvalcore/_recipe/recipe.py` and applying the regridding based on this? This would avoid double regridding.

Reply:
> How about checking the extension of the files in esmvalcore/_recipe/recipe.py and applying the regridding based on this? This would avoid double regridding.

Sounds good, could you give that a try?

Reply:
Implemented in https://github.com/ESMValGroup/ESMValCore/pull/2178/commits/e123244d16b99564c29a62e6d84f15568fcf6030